### PR TITLE
(feat) O3-4026 Add an icon for vitals and biometrics on the left nav

### PR DESCRIPTION
**Pull request by Munyua123**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Screenshots

image

image

### CODE REVIEW

**Comment by Munyua123** [comment]
@denniskigen I have made the changes on the icon using the Activity Icon.

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import last from 'lodash-es/last';
 import { useTranslation } from 'react-i18next';
 import { useLocation } from 'react-router-dom';
 import { ConfigurableLink } from '@openmrs/esm-framework';
+import { ActivityIcon } from '@openmrs/esm-framework';
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:7]: Can we consolidate these imports?

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 2]
```diff
@@ -4,6 +4,8 @@ import last from 'lodash-es/last';
...
+import styling from './dashboardextension.scss';
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:8]:
```suggestion
import styles from './dashboardextension.scss';
```

Can we rename this file to `dashboard-extension.scss` for consistency? https://o3-docs.openmrs.org/docs/coding-conventions#naming

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 3]
```diff
@@ -22,13 +24,18 @@ export const DashboardExtension = ({
   const location = useLocation();
   const navLink = useMemo(() => decodeURIComponent(last(location.pathname.split('/'))), [location.pathname]);
 
+  const renderIcon = title === 'Vitals & Biometrics' ? <ActivityIcon className={styling.icons} /> : null;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:27]:
```suggestion
  const renderIcon = title === 'Vitals & Biometrics' ? <ActivityIcon className={styles.icons} /> : null;
```

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 4]
```diff
@@ -22,13 +24,18 @@ export const DashboardExtension = ({
...
+
   return (
     <div key={path}>
       <ConfigurableLink
         className={classNames('cds--side-nav__link', { 'active-left-nav-link': path === navLink })}
         to={`${basePath}/${encodeURIComponent(path)}`}
       >
-        {t(title)}
+        <span className={styling.menu}>
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:35]:
```suggestion
        <span className={styling.menu}>
```

Ditto

File: packages/esm-patient-common-lib/src/dashboards/dashboardextension.scss [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.menu {
+  display: flex;
+  align-items: center;
+}
+
+.icons {
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/dashboardextension.scss:6]: Would `.icon` be more preferable here? It's just one icon, right?

**Review by denniskigen — COMMENTED** [review]
Thanks for making the changes, @Munyua123. It's close. I've left a few more recommendations.

**Comment by Munyua123** [comment]
@denniskigen I have made the changes that you had suggested. I have also made a change to use the switch case method, so that someone who is implementing more icons can just add the icons easily

**Comment by Munyua123** [comment]
@denniskigen  I have made the changes and added the icons for all of them but the immunizations tab. It seems that it has not been added to the openmrs framework and also there is none in the carbon react one. Also the icon i have used for the patient summary has not been added yet to the framework so i have used the one in the carbon react icons.

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 6]
```diff
@@ -22,13 +36,47 @@ export const DashboardExtension = ({
...
+      case 'Programs':
+        return <TableIcon className={styles.icon} />;
+      case 'Appointments':
+        return <EventScheduleIcon className={styles.icon} />;
+      default:
+        return null;
+    }
+  };
+
+  const renderIcon = menuIcon(title);
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:68]: Maybe an object lookup like the following is more preferable. It has less code and better type inference, and it should be straightforward to pop in more icons:

```suggestion
  type MenuTitle = keyof typeof MenuIcons;

  const MenuIcons = {
    Allergies: WarningIcon,
    Appointments: EventScheduleIcon,
    Attachments: DocumentAttachmentIcon,
    Conditions: ListCheckedIcon,
    Medications: MedicationIcon,
    Orders: ShoppingCartIcon,
    'Patient Summary': Report,
    Programs: TableIcon,
    Results: ChartAverageIcon,
    Visits: CalendarHeatMapIcon,
    'Vitals & Biometrics': ActivityIcon,
  } as const;

  const menuIcon = (title: MenuTitle) => {
    const Icon = MenuIcons[title];
    return Icon ? <Icon className={styles.icon} /> : null;
  };

  const renderIcon = menuIcon(title as MenuTitle);
```

**Comment by Munyua123** [comment]
@denniskigen The above is noted and will be implemented. I had a question will it be okay if i add the missing icons to the framework? or do i have to wait for directive on this

**Comment by denniskigen** [comment]
Yes, please go ahead and add them in [Core](https://github.com/openmrs/openmrs-esm-core).

**Comment by Munyua123** [comment]
@denniskigen I have added all  the icons , kindly review

**Comment by denniskigen** [comment]
@paulsonder @ciaranduffy could you give this a design review?

**Comment by paulsonder** [comment]
@denniskigen Works for me. We'll just need to coordinate on making sure these are reflected in the design docs.

File: packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx [hunk 7]
```diff
@@ -22,13 +37,40 @@ export const DashboardExtension = ({
   const location = useLocation();
   const navLink = useMemo(() => decodeURIComponent(last(location.pathname.split('/'))), [location.pathname]);
 
+  type MenuTitle = keyof typeof MenuIcons;
+
+  const MenuIcons = {
```
Comment (CynthiaKamau) [inline packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:42]: Thanks @Munyua123 for this, what will be the approach for adding icons for packages that do not live in the patient chart ?

Comment (Munyua123) [reply packages/esm-patient-common-lib/src/dashboards/DashboardExtension.tsx:42]: @CynthiaKamau The approach may be adding the icons in their respective packages.

**Comment by gracepotma** [comment]
Thank you **so much** @Munyua123!! I'm so grateful to you for working on this! These little icons signal a meaningful change in us making the app look a little less boring - and making it faster for providers to find the Nav page they want :)  

(1) Re. Immunizations: 
> @denniskigen I have made the changes and added the icons for all of them but the immunizations tab. It seems that it has not been added to the openmrs framework and also there is none in the carbon react one. Also the icon i have used for the patient summary has not been added yet to the framework so i have used the one in the carbon react icons.

@Munyua123 Please proceed with using the "Syringe" asset for Immunizations. 
This can be found [here](https://zeroheight.com/23a080e38/p/993dce-icons) under "Clinical" and looks like this: 
image


(2) Re. Patient Summary: 
The correct icon for Patient Summary is [here](https://zeroheight.com/23a080e38/p/993dce-icons) under "Records and documentation" here and looks like this: 
image


(3) Re. Programs: 
The correct icon for Programs is [here](https://zeroheight.com/23a080e38/p/993dce-icons) under "Records and documentation" here and looks like this: 
image


Hope that helps - thank you again; once those are done, can't wait to see this in the next release :D

**Comment by Munyua123** [comment]
@gracepotma Thank you for providing the above feedback, I was able to add the syringe and patient summary icon to the openMRS framework, but it seems the programs icon has not yet been added, i will add it to the framework, then update the patient chart.

**Comment by Munyua123** [comment]
@gracepotma @denniskigen Here is the final look of the patient chart menu. 
image
Kindly review and let me know of any potential changes.

**Comment by CynthiaKamau** [comment]
Thanks @Munyua123 , please fix the failing build

**Comment by Munyua123** [comment]
@CynthiaKamau Kindly try again it should pass now

**Pull request closed without merging**